### PR TITLE
Add tooltip management on check all checkbox

### LINKED ISSUE #227: [FEATURE] Checkbox select all tooltip
Is it possible to add an @input to set text of a MatTooltip on the checkbox to select/unselect all ?

### PR DESCRIPTION
**Pull request by arucar**
Update readme, add new exemple based on exemple number 6, add two intpu, one to manage text of the tooltip, another to manage the postion.

i dont change the version number because i don't know if it's up to me to change that.

hope it will be good for you...

unit test ok
buil --prod ok

fixes #227

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,6 +144,12 @@ Furthermore, it provides the following inputs:
   /** select all checkbox indeterminate state */
   @Input() toggleAllCheckboxIndeterminate = false;
 
+  /** Display a message in a tooltip on the checkbox */
+  @Input() toggleAllCheckboxTooltipMessage = '';
+
+  /** Define the position of the tooltip. Available options: 'left' | 'right' | 'above' | 'below' | 'before' | 'after'. */
+  @Input() toogleAllCheckboxTooltipPosition = 'below';
```
Comment (macjohnny) [inline README.md:151]:
```suggestion
  @Input() toogleAllCheckboxTooltipPosition: 'left' | 'right' | 'above' | 'below' | 'before' | 'after' = 'below';
```

File: README.md [hunk 2]
```diff
@@ -144,6 +144,12 @@ Furthermore, it provides the following inputs:
   /** select all checkbox indeterminate state */
   @Input() toggleAllCheckboxIndeterminate = false;
 
+  /** Display a message in a tooltip on the checkbox */
+  @Input() toggleAllCheckboxTooltipMessage = '';
+
+  /** Define the position of the tooltip. Available options: 'left' | 'right' | 'above' | 'below' | 'before' | 'after'. */
```
Comment (macjohnny) [inline README.md:150]:
```suggestion
  /** Define the position of the tooltip on the toggle-all checkbox. */
```

File: src/app/mat-select-search/mat-select-search.component.html [hunk 3]
```diff
@@ -12,6 +12,9 @@
                 class="mat-select-search-toggle-all-checkbox"
                 [checked]="toggleAllCheckboxChecked"
                 [indeterminate]="toggleAllCheckboxIndeterminate"
+                [matTooltip]="toggleAllCheckboxTooltipMessage"
+                matTooltipClass="tooltip"
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.html:16]:
```suggestion
                matTooltipClass="ngx-mat-select-search-toggle-all-tooltip"
```

File: src/app/mat-select-search/mat-select-search.component.scss [hunk 4]
```diff
@@ -116,3 +116,7 @@ $mat-option-height: 3em;
   padding-left: 16px;
   padding-bottom: 2px;
 }
+
+::ng-deep .mat-tooltip.tooltip {
+  font-size: 0.8em;
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.scss:121]: why is this needed? the user could adapt the font size himself

File: src/app/mat-select-search/mat-select-search.component.ts [hunk 5]
```diff
@@ -171,6 +171,12 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
   /** select all checkbox indeterminate state */
   @Input() toggleAllCheckboxIndeterminate = false;
 
+  /** Display a message in a tooltip on the checkbox */
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts:174]:
```suggestion
  /** Display a message in a tooltip on the toggle-all checkbox */
```

File: src/app/mat-select-search/mat-select-search.component.ts [hunk 6]
```diff
@@ -171,6 +171,12 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
...
+  @Input() toggleAllCheckboxTooltipMessage = '';
+
+  /** Define the position of the tooltip. Available options: 'left' | 'right' | 'above' | 'below' | 'before' | 'after'. Default: 'below' */
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts:177]:
```suggestion
  /** Define the position of the tooltip on the toggle-all checkbox. */
```

File: src/app/mat-select-search/mat-select-search.component.ts [hunk 7]
```diff
@@ -171,6 +171,12 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
...
+  @Input() toogleAllCheckboxTooltipPosition = 'below';
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts:178]:
```suggestion
  @Input() toogleAllCheckboxTooltipPosition: 'left' | 'right' | 'above' | 'below' | 'before' | 'after' = 'below';
```

File: README.md [hunk 8]
```diff
@@ -144,6 +144,12 @@ Furthermore, it provides the following inputs:
   /** select all checkbox indeterminate state */
   @Input() toggleAllCheckboxIndeterminate = false;
 
+  /** Display a message in a tooltip on the checkbox */
```
Comment (macjohnny) [inline README.md:147]:
```suggestion
  /** Display a message in a tooltip on the toggle-all checkbox */
```

**Comment by arucar** [comment]
changes are ok for me, i need to click on commit suggestion for all changes ?

**Comment by macjohnny** [comment]
@arucar yes, you can apply the changes online, or apply them by hand in your branch

File: src/app/mat-select-search/mat-select-search.component.scss (hunk 4 above)
Comment (arucar) [reply src/app/mat-select-search/mat-select-search.component.scss:121]: hum yes maybe i need to test it to update the css directly in the exemple and not in the component

**Comment by arucar** [comment]
i have make  and test the changes for CSS + add information about the css class in the Readme

**Comment by macjohnny** [comment]
@arucar thanks for your contribution!

**Comment by macjohnny** [comment]
@arucar thanks for your contribution, it was released in version `2.2.0`

**Comment by arucar** [comment]
Yes I see that, thank you 🎉
